common/lib/dependabot/shared_helpers.rb

```diff
@@ -193,11 +193,20 @@ def self.run_helper_subprocess(command:, function:, args:, env: nil,
           trace: response["trace"]
         )
       rescue JSON::ParserError
-        raise HelperSubprocessFailed.new(
-          message: stdout || "No output from command",
-          error_class: "JSON::ParserError",
-          error_context: error_context
-        )
+        # Handle case when stdout is empty but stderr is not
+        if stdout.strip.empty? && !stderr.strip.empty?
+          raise HelperSubprocessFailed.new(
+            message: stderr,
+            error_class: "StandardError",
+            error_context: error_context
+          )
+        else
+          raise HelperSubprocessFailed.new(
+            message: stdout || "No output from command",
```

Comment:
seems this is now redundant?

Reply:
No it is not. Just made changes. If we don't have stdout then we are using stderr.